Userland/Libraries/LibC/malloc.cpp

```diff
@@ -243,21 +277,27 @@ static void* malloc_impl(size_t size, CallerWillInitializeMemory caller_will_ini
     }
 
     --block->m_free_chunks;
-    void* ptr = block->m_freelist;
+    if (!ptr)
+        ptr = find_aligned_entry(*block);
+
     if (ptr) {
         block->m_freelist = block->m_freelist->next;
     } else {
-        ptr = block->m_slot + block->m_next_lazy_freelist_index * block->m_size;
-        block->m_next_lazy_freelist_index++;
+        void* candidate_ptr = block->m_slot + block->m_next_lazy_freelist_index * block->m_size;
+        if ((reinterpret_cast<uintptr_t>(candidate_ptr) & (alignment - 1)) == 0) {
+            ptr = candidate_ptr;
+            block->m_next_lazy_freelist_index++;
+        }
     }
+    
```

Comment:
@gunnarbeutner do the changes here seem reasonable to you when considering 343882943166abf4542a0e6643baf9d80806e944?